src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.db.StorageImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.FileReader;
+import core.basesyntax.service.FileWriter;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.TransactionParser;
+import core.basesyntax.service.impl.FileReaderImpl;
+import core.basesyntax.service.impl.FileWriterImpl;
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import core.basesyntax.service.impl.ShopServiceImpl;
+import core.basesyntax.service.impl.TransactionParserImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.BalanceOperation;
+import core.basesyntax.strategy.impl.OperationStrategyImpl;
+import core.basesyntax.strategy.impl.PurchaseOperation;
+import core.basesyntax.strategy.impl.ReturnOperation;
+import core.basesyntax.strategy.impl.SupplyOperation;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+        if (args.length != 2) {
+            System.out.println("Please provide two filepaths - for reading and writing - "
+                    + "as arguments.");
+            return;
+        }
+        Map<Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(Operation.BALANCE, new BalanceOperation());
+        operationHandlers.put(Operation.PURCHASE, new PurchaseOperation());
+        operationHandlers.put(Operation.RETURN, new ReturnOperation());
+        operationHandlers.put(Operation.SUPPLY, new SupplyOperation());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+
+        Storage storage = new StorageImpl();
+        ShopService shopService = new ShopServiceImpl(operationStrategy, storage);
+
+        FileReader fileReader = new FileReaderImpl();
+        String readPath = args[0];
+        List<String> inputReport = fileReader.read(readPath);
+
+        TransactionParser transactionParser = new TransactionParserImpl();
+        List<FruitTransaction> transactions = transactionParser.parse(inputReport);
+
+        shopService.process(transactions);
```

Comment:
The `ShopService` should be called in the `main()` method with `transactions` as a parameter, but there seems to be a tight coupling between `ShopService` and `OperationStrategy`. Make sure to decouple them according to the SOLID principles and the checklist requirements.